core/src/main/java/fj/data/List.java

```diff
@@ -563,7 +562,57 @@ public final void foreachDoEffect(final Effect1<A> f) {
     return bind(c);
   }
 
-  /**
+    /**
+     * Traverse through the List with a might-fail function.
+     *
+     * @param f The function that might fail.
+     * @return  none if applying f fails to any element of the list or f mapped list in some .
+     */
+    public <B> Option<List<B>> traverseOption(final F<A, Option<B>> f) {
+        F2<A, Option<List<B>>, Option<List<B>>> f2 = (a, as) -> map2Option(List::cons, f.f(a), as);
+        return foldRight(f2, Option.some(List.<B>nil()));
+    }
+
+    private static <A, B, C> Option<List<C>> map2Option(F2<A, List<B>, List<C>> f, final Option<A> opt1, final Option<List<B>> opt2) {
+        return opt1.bind(o1 -> opt2.map(o2 -> f.f(o1, o2)));
+    }
+
```

Comment:
I would suggest to rename these variables, e.g. on line 372 `as` represent `Option<List<B>>`.  Why not use `obs` or `olb` here?